VirtualFCS/Powertrains/BatteryPowerTrain2.mo

```diff
@@ -0,0 +1,55 @@
+within VirtualFCS.Powertrains;
+
+model BatteryPowerTrain2
+  extends VehicleInterfaces.EnergyStorage.Interfaces.Base;
+  
+  parameter Real m_powertrain(unit = "kg") = 100 + 50;
+  parameter Real V_HV_Bus(unit = "V") = 343 "Voltage of the HV Bus";
+  // H2 Subsystem Paramters
+  parameter Real V_tank_H2(unit = "m3") = 0.13 "H2 tank volume";
+  parameter Real p_tank_H2(unit = "Pa") = 3500000 "H2 tank initial pressure";
+  // Fuel Cell Stack Paramters
+  parameter Real m_FC_stack(unit = "kg") = 14.3 "FC stack mass";
+  parameter Real L_FC_stack(unit = "m") = 0.255 "FC stack length";
+  parameter Real W_FC_stack(unit = "m") = 0.760 "FC stack length";
+  parameter Real H_FC_stack(unit = "m") = 0.060 "FC stack length";
+  parameter Real vol_FC_stack(unit = "m3") = L_FC_stack * W_FC_stack * H_FC_stack "FC stack volume";
+  parameter Real V_rated_FC_stack(unit = "V") = 57.9 "FC stack maximum operating voltage";
+  parameter Real I_rated_FC_stack(unit = "A") = 300 "FC stack minimum operating voltage";
+  parameter Real i_L_FC_stack(unit = "A") = 3 * I_rated_FC_stack "FC stack maximum limiting current";
+  parameter Real I_nom_FC_stack(unit = "A") = 0.25 * I_rated_FC_stack "FC stack maximum limiting current";
+  parameter Real N_FC_stack(unit = "1") = floor(V_rated_FC_stack / 0.6433) "FC stack number of cells";
+  // Battery Pack Parameters
+  parameter Real m_bat_pack(unit = "kg") = 100 "Mass of the pack";
+  parameter Real L_bat_pack(unit = "m") = 0.6 "Battery pack length";
+  parameter Real W_bat_pack(unit = "m") = 0.45 "Battery pack width";
+  parameter Real H_bat_pack(unit = "m") = 0.1 "Battery pack height";
+  parameter Real Cp_bat_pack(unit = "J/(kg.K)") = 1000 "Specific Heat Capacity";
+  parameter Real V_min_bat_pack(unit = "V") = 37.5 "Battery pack minimum voltage";
+  parameter Real V_nom_bat_pack(unit = "V") = 48 "Battery pack nominal voltage";
+  parameter Real V_max_bat_pack(unit = "V") = 54.75 "Battery pack maximum voltage";
+  parameter Real C_bat_pack(unit = "A.h") = 2700 "Battery pack nominal capacity";
+  parameter Real SOC_init = 0.5 "Battery pack initial state of charge";
+  VirtualFCS.Electrochemical.Battery.BatterySystem batterySystem(C_bat_pack = 200, SOC_init = 0.99, V_max_bat_pack = 300, V_min_bat_pack = 0) annotation(
+    Placement(visible = true, transformation(origin = {0, -70}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  VirtualFCS.Electrical.DCConverter converter(vDCref = V_HV_Bus) annotation(
+    Placement(visible = true, transformation(origin = {0, 8}, extent = {{-10, -10}, {10, 10}}, rotation = -90)));
+  Modelica.Blocks.Interfaces.RealOutput y annotation(
+    Placement(visible = true, transformation(origin = {-48, 14}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {-48, 14}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
```

Comment:
@dietmarw - How would you recommend combining an interface like `Modelica.Blocks.Interfaces.RealOutput` with a type like `Modelica.Units.SI.AbsolutePressure` to specify `quantity`, `unit`, `displayUnit`, `min`, `nominal`, etc. by just referring to the type identifier `Modelica.Units.SI.AbsolutePressure` without copying all parameters into the instantiation like this:
```Modelica
Modelica.Blocks.Interfaces.RealOutput y(quantity="Pressure", unit="Pa", displayUnit="bar", min=0.0, nominal=1e5);
```
Copying the other way is not much better:
```Modelica
output Modelica.Units.SI.AbsolutePressure y annotation(...);
```
Then all the graphical annotations from the interface `Modelica.Blocks.Interfaces.RealOutput` must be copied.
I'm seeking a way to combine the interface identifier with the type identifier. I would expect something like this should work, but it doesn't seem to compile (Syntax Error):
```Modelica
Modelica.Blocks.Interfaces.RealOutput y(type=Modelica.Units.SI.AbsolutePressure);
```
I use many combinations of interfaces and types, and want to combine existing declarations in an efficient way to avoid code duplication.

Reply:
@kvid Ideally having to set the unit of an output should not be necessary and the tool should be able to deduce the type based on the (typed) variables and parameters involved in calculating the output's signal. Now that is the ideal solution which of course also needs support from the tool. OMEdit will check for unit conflicts (when you press "check model") but will (as of now) not go as far as propagate units so that for example plots show the units. For these cases you need to specify the unit manually but you do not really need to go as far as also specifying `quantity` and `nominal`. 

So for now, the recommended way is 
```mo
Modelica.Blocks.Interfaces.RealOutput y(unit="Pa", displayUnit="bar");
```
See also https://github.com/modelica/ModelicaStandardLibrary/pull/2197/files  as an example how the MSL did it.